# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
# Commit: 30340541c9a1
# Full Hash: 30340541c9a17e21f1c8f7e65850afaa5e3355db
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-03-06 16:15:33
# Regressor Bug: 1817724
# File Overlap Count: 3
# Description:
#   Bug 1817724 - Crash in case VideoCaptureAVFoundation is destroyed unexpectedly. r=karlt,webrtc-reviewers,ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171470
# ==============================================================================

diff -r 46e97efba193 -r 30340541c9a1 dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Mon Mar 06 06:58:34 2023 +0000
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Mon Mar 06 07:41:28 2023 +0000
@@ -114,6 +114,7 @@
 
 VideoCaptureAvFoundation::~VideoCaptureAvFoundation() {
   // Must block until capture has fully stopped, including async operations.
+  MOZ_RELEASE_ASSERT(mOkToDestroy);
   StopCapture();
 }
 